iceaxe/__tests__/conftest.py

```diff
@@ -1,84 +1,300 @@
+                    name=container_name,
+                    detach=True,
+                    environment={
+                        "POSTGRES_USER": pg_user,
+                        "POSTGRES_PASSWORD": pg_password,
+                        "POSTGRES_DB": pg_db,
+                    },
+                    ports={"5432/tcp": port},
+                    remove=True,
+                )
+            except Exception as inner_e:
+                raise RuntimeError(
+                    f"Failed to start PostgreSQL container with random port: {inner_e}"
+                ) from e
+        else:
+            raise RuntimeError(f"Failed to start PostgreSQL container: {e}")
+
+    # Wait for PostgreSQL to be ready
+    max_retries = 30
+    retry_interval = 1
+    for i in range(max_retries):
+        container.reload()  # Refresh container status
+        if container.status != "running":
+            raise RuntimeError(f"Container failed to start: {container.status}")
+
+        # Try to connect to PostgreSQL
+        try:
+            conn = socket.create_connection(("localhost", port), timeout=1)
+            conn.close()
+            break
+        except (socket.error, ConnectionRefusedError):
+            if i == max_retries - 1:
+                container.stop()
+                raise RuntimeError("Failed to connect to PostgreSQL container")
+            time.sleep(retry_interval)
+
+    # Wait a bit more to ensure PostgreSQL is fully initialized
+    time.sleep(2)
+
+    # Yield connection details
+    connection_info = {
+        "host": "localhost",
+        "port": port,
+        "user": pg_user,
+        "password": pg_password,
+        "database": pg_db,
+    }
+
+    yield connection_info
+
+    # Cleanup: stop the container
+    try:
+        container.stop()
+    except Exception as e:
+        logger.warning(f"Failed to stop container: {e}")
+
 
 @pytest_asyncio.fixture
-async def db_connection():
+async def db_connection(docker_postgres):
+    """
+    Create a database connection using the PostgreSQL container.
+    """
     conn = DBConnection(
         await asyncpg.connect(
-            host="localhost",
-            port=5438,
-            user="iceaxe",
-            password="mysecretpassword",
-            database="iceaxe_test_db",
+            host=docker_postgres["host"],
+            port=docker_postgres["port"],
+            user=docker_postgres["user"],
+            password=docker_postgres["password"],
+            database=docker_postgres["database"],
         )
     )
 
     # Drop all tables first to ensure clean state
-    await conn.conn.execute("DROP TABLE IF EXISTS artifactdemo CASCADE")
-    await conn.conn.execute("DROP TABLE IF EXISTS userdemo CASCADE")
-    await conn.conn.execute("DROP TABLE IF EXISTS complexdemo CASCADE")
-    await conn.conn.execute("DROP TABLE IF EXISTS article CASCADE")
+    await conn.conn.execute("DROP TABLE IF EXISTS artifactdemo CASCADE", timeout=30.0)
```

Comment:
The query string here looks to be shared for all of the tables. Probably makes sense to either do a query of the REGISTRY var to get the tables that are dynamically registered from their classes, or as a simpler case just loop over a list of known table names.